### ISSUE TITLE
Calling exit within an implicit subshell should be flagged as a warning

### ISSUE THREAD

**Issue by srcshelton** [koalaman/shellcheck#751]
Implicit sub-shells in bash are still IMHO the greatest source of scripting errors:

Please flag a warning for:

```
input | while read -r variable; do
    [[ condition ]] || exit 1
done
echo ${?}
```

... due to the implicit sub-shell generated by the pipe symbol, any script containing this sequence of commands will not, as anticipated, exit with status 1, but will instead print '`1`' and continue.

This is almost certainly not what is intended by the author...

If the parser is flexible enough, this could even be flagged as an error if the return-status of the while block is never checked, strongly indicating that the `exit` was intended to actually exit.

**Comment by ijustlovemath** [comment]
Hi, I just wanted to bump this issue! I have a simple example here:

Broken:
```bash
function vpn_connected () {
    ip route | while read line
    do
        if [[ "$line" == "0.0.0.0/1"*"dev tun0" ]]; then
            true
            return
        fi
    done
    false
    return
}

if vpn_connected; then
     echo "connected";
else
    echo "not connected";
fi

```
Working:
```bash
function vpn_connected () {
    while read line
    do
        if [[ "$line" == "0.0.0.0/1"*"dev tun0" ]]; then
            true
            return
        fi
    done < <(ip route)
    false
    return
}

if vpn_connected; then
     echo "connected";
else
    echo "not connected";
fi

```
